include/matx/operators/cast.h

```diff
@@ -108,8 +108,80 @@ namespace matx
           return op_.Size(dim);
         }
     };
-  }   
 
+    template <typename T1, typename T2, typename NewType>
+      class ComplexCastOp : public BaseOp<ComplexCastOp<T1, T2, NewType>>
+    {
+      private:
+        typename base_type<T1>::type real_op_;
+        typename base_type<T2>::type imag_op_;
+
+      public:
+        using matxop = bool;
+        using scalar_type = NewType;
+        static_assert(!is_complex_v<T1> && !is_complex_half_v<T1>, "T1 input operator cannot be complex");
+        static_assert(!is_complex_v<T2> && !is_complex_half_v<T2>, "T2 input operator cannot be complex");
+        static_assert(is_complex_v<NewType> || is_complex_half_v<NewType>, "ComplexCastOp output type should be complex");
+
+	      __MATX_INLINE__ std::string str() const { return as_type_str<NewType>() + "(" + real_op_.str() + "," + imag_op_.str() + ")"; }
+        __MATX_INLINE__ ComplexCastOp(T1 real_op, T2 imag_op) : real_op_(real_op), imag_op_(imag_op) {
+          static_assert(detail::get_rank<T1>() == detail::get_rank<T2>(), "rank of real and imaginary operators must match");
+          for (auto i = 0; i < detail::get_rank<T1>(); i++) {
+            if (real_op_.Size(i) != imag_op_.Size(i)) {
+              MATX_THROW(matxInvalidSize, "ComplexCastOp: sizes of input operators must match in all dimensions");
+            }
+          }
+        };
+
+        template <typename... Is>
+        __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ decltype(auto) operator()(Is... indices) const
```

Comment:
Do we not want `auto` here?

Reply:
Yes, I believe so. I copy-pasted that from CastOp, so can change both. I'll check what our other `operator()(...) const` return types are as well.

Reply:
I changed this case to auto, but we may want a follow-up PR to revert other operators to auto.